### DISCUSSION TITLE
`ava` override `require.main`, what should i do?

### DISCUSSION THREAD

**Discussion by Airkro** [avajs/ava#2736] [Q&A]
```js
// sample.cjs
module.exports = require.main;
```

```js
// run.mjs
import sample from './sample.cjs';
console.log(sample);
```

```sh
# shell
$ node run.mjs
> undefined
```

```js
// test.mjs
import test from 'ava';
import sample from './sample.cjs';

test('name', (t) => {
  t.is(sample, undefined);
});
```

```sh
# shell
$ npx ava test.mjs
>
> Difference:
>
> - Module {
> -   filename: 'I:\\node\\fs-chain\\node_modules\\ava\\lib\\worker\\subprocess.js'
> - }
> + undefined
>
> 1 test failed
```

**Comment by novemberborn** [comment]
`ava: fail` isn't an AVA output. What does it print? Best to run with `-v` for verbose output.

How did you install AVA? Your invocation implies a global installation. If you install locally and run with `npx ava`, does the outcome change?

My best guess is that AVA errors for a different reason. There shouldn't be a difference in Node.js behavior, but you haven't shared your AVA version and configuration, or Node.js version, so perhaps there's something there.

**Reply by Airkro** [reply]
sorry, it is locally run,

`ava: fail` isn't the output, it just means the test will fail.

I have updated the code sample.
